extensions/standard-processors/utils/JoltUtils.cpp

```diff
@@ -0,0 +1,1134 @@
+      }
+      ++ch_it;
+    }
+    if (!isEnd()) {
+      if (*ch_it == '.') {
+        type = Spec::MemberType::FIELD;
+      } else if (*ch_it == '[') {
+        type = Spec::MemberType::INDEX;
+      } else {
+        return nonstd::make_unexpected(fmt::format("Unexpected destination delimiter '{}' in {} at {}", *ch_it, ctx.path(), std::distance(begin, ch_it)));
+      }
+      ++ch_it;
+      if (ch_it == end) {
+        if (type == Spec::MemberType::FIELD) {
+          return nonstd::make_unexpected(fmt::format("Unterminated member in {} at {}", ctx.path(), std::distance(begin, ch_it)));
+        } else {
+          return nonstd::make_unexpected(fmt::format("Unterminated indexed member in {} at {}", ctx.path(), std::distance(begin, ch_it)));
+        }
+      }
+    }
+  }
+
+  return std::pair<Spec::Destination, Spec::It>{result, ch_it};
+}
+
+Spec::Destinations parseDestinations(const Spec::Context& ctx, const rapidjson::Value& val) {
+  Spec::Destinations res;
+  if (val.IsNull()) {
+    return res;
+  }
+  if (val.IsArray()) {
+    for (rapidjson::SizeType i = 0; i < val.GetArray().Size(); ++i) {
+      auto& item = val.GetArray()[i];
+      if (!item.IsString()) {
+        throw Exception(GENERAL_EXCEPTION, fmt::format("Expected a string or array of strings at '{}/{}'", ctx.path(), i));
+      }
+      if (auto dst = parseDestination(ctx, item.GetString(), item.GetString() + item.GetStringLength())) {
+        if (dst->second != item.GetString() + item.GetStringLength()) {
+          throw Exception(GENERAL_EXCEPTION, fmt::format("Failed to fully parse destination at '{}/{}'", ctx.path(), i));
+        }
+        res.push_back(std::move(dst->first));
+      } else {
+        throw Exception(GENERAL_EXCEPTION, fmt::format("Failed to parse destination at '{}/{}': {}", ctx.path(), i, dst.error()));
+      }
+    }
+  } else {
+    if (!val.IsString()) {
+      throw Exception(GENERAL_EXCEPTION, fmt::format("Expected a string or array of strings at '{}'", ctx.path()));
+    }
+    if (auto dst = parseDestination(ctx, val.GetString(), val.GetString() + val.GetStringLength())) {
+      if (dst->second != val.GetString() + val.GetStringLength()) {
+        throw Exception(GENERAL_EXCEPTION, fmt::format("Failed to fully parse destination at '{}'", ctx.path()));
+      }
+      res.push_back(std::move(dst->first));
+    } else {
+      throw Exception(GENERAL_EXCEPTION, fmt::format("Failed to parse destination at '{}': {}", ctx.path(), dst.error()));
+    }
+  }
+  return res;
+}
+
+std::optional<std::string> jsonValueToString(const rapidjson::Value& val) {
+  if (val.IsString()) {
+    return std::string{val.GetString(), val.GetStringLength()};
+  }
+  if (val.IsUint64()) {
+    return std::to_string(val.GetUint64());
+  }
+  if (val.IsInt64()) {
+    return std::to_string(val.GetInt64());
+  }
+  if (val.IsDouble()) {
+    return std::to_string(static_cast<int64_t>(val.GetDouble()));
```

Comment:
this looks weird: do we really want to `static_cast` the double to an `int64_t`?

Reply:
it is weird, but the reference impl does this conversion